config/config.go

```diff
@@ -76,6 +78,29 @@ func NewConfig() *Configuration {
 	return c
 }
 
+// NewFromValues returns a new *config.Configuration instance as if it had
+// been read from the .gitconfig specified by "gitconfig" parameter.
+//
+// NOTE: this method should only be called during testing.
+func NewFromValues(gitconfig map[string]string) *Configuration {
+	config := &Configuration{
```

Comment:
You could actually put this in config_test.go to enforce the 'only testing' requirement.